### PR TITLE
refactor cocoapods validator to detect broken install

### LINKED ISSUE #14293: Misleading flutter doctor message about cocoapods
Flutter doctor was giving me this message:

```
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without resolving iOS dependencies with CocoaPods, plugins will not work on iOS.
        For more info, see https://flutter.io/platform-plugins
      To install:
        brew install cocoapods
        pod setup
```

When executing `brew install cocoapods` as suggested, I got this message:

```
Error: cocoapods 1.3.1 is already installed
To upgrade to 1.4.0, run `brew upgrade cocoapods`
```

Do we need to change the doctor message and detect that it is maybe just out of date?

/cc @xster @cbracken

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

When CocoaPods are installed they are dependent on a specific version of ruby. If the `pod` script is invoked while a different version of ruby is set as the default, it will fail with an error message. The most common cause for this is upgrading macOS, which changes the system ruby version, however it can happen in other situations (the most annoying of which is the RVM situation). `sudo gem install cocoapods` will usually fix this, but I added a link to the issue which has various variations of fixes.

## Related Issues

Fixes [this issue](https://github.com/flutter/flutter/issues/14293).

## Tests

I added the following tests:

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter_tools/lib/src/macos/cocoapods_validator.dart [hunk 1]
```diff
@@ -34,6 +34,11 @@ class CocoaPodsValidator extends DoctorValidator {
         status = ValidationType.missing;
         messages.add(ValidationMessage.error(
             userMessages.cocoaPodsMissing(noCocoaPodsConsequence, cocoaPodsInstallInstructions)));
+      } else if (cocoaPodsStatus == CocoaPodsStatus.brokenInstall) {
+        status = ValidationType.missing;
+        messages.add(ValidationMessage.error(
+            userMessages.cocoaPodsBrokenInstall(brokenCocoaPodsConsequence, cocoaPodsInstallInstructions)));
```
Comment (blasten) [inline packages/flutter_tools/lib/src/macos/cocoapods_validator.dart]: minor nit: considering making `cocoaPodsBrokenInstall` if the consequence and instructions don't change.

**Review by blasten — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/macos/cocoapods.dart [hunk 2]
```diff
@@ -29,6 +29,11 @@ const String unknownCocoaPodsConsequence = '''
   Flutter is unable to determine the installed CocoaPods's version.
   Ensure that the output of 'pod --version' contains only digits and . to be recognized by Flutter.''';
 
+const String brokenCocoaPodsConsequence = '''
+  You appear to have CocoaPods installed but it is not working.
+  This can happen if the version of ruby that CocoaPods was installed with is different from the one being used to invoke it.
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/macos/cocoapods.dart]: Ruby capitalized

**Review by jmagman — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/lib/src/macos/cocoapods.dart (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/macos/cocoapods.dart]: done

**Pull request merged**